lib/AI-Agent.xml

```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
 <ApplicationInsightsAgent>
-    <AgentLogger>INFO</AgentLogger>
+    <AgentLogger>DEBUG</AgentLogger>
```

Comment:
can we check with Platform Engineering to see that they are ok with this? not sure what the conventions are

Reply:
In fact, by having the root level logging at INFO level, we are writing no debug logs at all, by default.
By configuring only a specific class to log at DEBUG level, we are writing logs in debug level only in that class, if it contains some.

The AI agent is sending anything it receives at or above DEBUG, but the resultant data pushed to Azure is only INFO Logs + any specific debug log. This is actually a nicer way of using the log levels, as you'd also appreciate.

Re sensitive info exposures, you're right. But that concern already applies to all logs.
One can write sensitive data in a log relying on that it's not visible anywhere, but then later config changes may expose such data. So, we are already assuring no logging of any sensitive data at all.

If we use only Info Logs, it also prevents us from use of debug logs in many many libraries that come with built-in logging at debug level. In the case of a need for, let's say, a debugging with help of a  Spring library logs, we will have to set the log level for a specific Spring class or package, but the current config will not allow the specific Spring debug log to appear on AI Resources.

Obviously, pushing to Azure debug logs is not something we can do arbitrarily, as it does come with its costs, but when we enable a debug log, we do so by expressly knowing and managing the fact that this is a temporary and very limited scope of debug logging for a good reason - which happens very rarely, like in this situation where we need some genuinely 'debug' logs.

Well, in short,
The idea is, actually, to be able to preserve the debug logging capability in the full stack of the app at a class-level granularity if and only when needed - and disabled by default by having root log level at INFO.

Reply:
this is just the agent logger, unless you are debugging something in the app insights agent I would be very surprised that you would need to change this.

Reply:
That's even better. So, it means the actual log level pushed to Azure is defined solely at application properties - nothing managed at AI Agent?

Reply:
Its configured here:
https://github.com/hmcts/java-logging/blob/master/src/main/resources/logback.xml#L5